app/api/entities/managerFunctions.ts

```diff
@@ -0,0 +1,219 @@
+          )
+      )
+      .map((file: FileType) => ({
+        _id: file._id!.toString(),
+        originalname: file.originalname,
+      }));
+
+  const renamedAttachments = entity.attachments ? process(entity.attachments) : [];
+
+  const renamedDocuments = entity.documents ? process(entity.documents) : [];
+
+  return { renamedAttachments, renamedDocuments };
+};
+
+const processFiles = async (
+  entity: EntityWithFilesSchema,
+  updatedEntity: EntityWithFilesSchema,
+  fileAttachments: FileType[] | undefined,
+  documentAttachments: FileType[] | undefined
+) => {
+  const { attachments, documents } = await prepareNewFiles(
+    entity,
+    updatedEntity,
+    fileAttachments,
+    documentAttachments
+  );
+
+  if (entity._id && (entity.attachments || entity.documents)) {
+    const entityFiles: WithId<FileType>[] = await filesAPI.get(
+      { entity: entity.sharedId, type: { $in: ['attachment', 'document'] } },
+      '_id, originalname, type'
+    );
+
+    await updateDeletedFiles(entityFiles, entity, 'attachment');
+    await updateDeletedFiles(entityFiles, entity, 'document');
+
+    const { renamedAttachments, renamedDocuments } = filterRenamedFiles(entity, entityFiles);
+
+    attachments.push(...renamedAttachments);
+    documents.push(...renamedDocuments);
+  }
+
+  return { proccessedAttachments: attachments, proccessedDocuments: documents };
+};
+
+const bindAttachmentToMetadataProperty = (
+  _values: MetadataObjectSchema[],
+  attachments: FileType[]
+) => {
+  const values = _values;
+  if (_values[0].attachment !== undefined) {
+    values[0].value = attachments[_values[0].attachment]
+      ? `/api/files/${attachments[_values[0].attachment].filename}`
+      : '';
+  }
+  return values;
+};
+
+const handleAttachmentInMetadataProperties = (
+  entity: EntityWithFilesSchema,
+  attachments: FileType[]
+) => {
+  Object.entries(entity.metadata || {}).forEach(([_property, _values]) => {
+    if (_values && _values.length) {
+      const values = bindAttachmentToMetadataProperty(_values, attachments);
+      delete values[0].attachment;
+    }
+  });
+
+  return entity;
+};
+
+const saveFiles = async (
+  attachments: FileType[],
+  documents: FileType[],
+  entity: ClientEntitySchema,
+  socketEmiter?: Function
+) => {
+  const saveResults: string[] = [];
+
+  const { documentsToProcess = [], documentsToSave = [] } = groupBy(documents, document =>
+    document._id ? 'documentsToSave' : 'documentsToProcess'
+  );
+
+  const filesToSave = [...attachments, ...documentsToSave];
+
+  await Promise.all(
+    filesToSave.map(async file => {
+      try {
+        await filesAPI.save(file, false);
+      } catch (e) {
+        errorLog.error(prettifyError(e));
+        saveResults.push(`Could not save file/s: ${file.originalname}`);
+      }
+    })
+  );
+
+  if (documentsToProcess.length) {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    Promise.allSettled(
```

Comment:
Since there is an eslint disable for it, I imagine that is the case, but let me double check it: this is intentionally not awaited? If yes, why? I'm asking since every other usage of the processDocument function I could find is awaited.

Reply:
That's right @LaszloKecskes, we are intentionally leaving the promise floating because we don't want the request to wait until the documents are processed because that could increase significative the response time at 'saving entities' when there are several documents to save. We are updating the state of the documents through the socket when they are completed. We agreed that behavior in a standup meeting after the problem was discussed. The main difference with other places where we are waiting for processDocument function is that there we are processing only one document at a time.

Reply:
Thanks for the explanation!